crates/toolchain/tests/tests/transpiler_tests.rs

```diff
@@ -123,6 +127,35 @@ fn test_rv32im_aot_pure_runtime(elf_path: &str) -> Result<()> {
     Ok(())
 }
 
+
+#[test_case("tests/data/rv32im-exp-from-as")]
+fn test_rv32im_aot_pure_runtime_with_path(elf_path: &str) -> Result<()> {
+    let elf = get_elf(elf_path)?;
+    let exe = VmExe::from_elf(
+        elf,
+        Transpiler::<F>::default()
+            .with_extension(Rv32ITranspilerExtension)
+            .with_extension(Rv32MTranspilerExtension)
+            .with_extension(Rv32IoTranspilerExtension),
+    )?;
+
+    let manifest_dir = env!("CARGO_MANIFEST_DIR");        
+    let config = Rv32ImConfig::default();
+    let executor = VmExecutor::new(config.clone())?;
+
+    let interpreter = executor.instance(&exe)?;
+    let interp_state = interpreter.execute(vec![], None)?;
+
+    let asm_name = String::from("asm_run");
+    let mut aot_instance = executor.aot_instance_with_asm_name(&exe, &asm_name)?;
```

Comment:
it's a little nicer if you have something like `aot_instance.with_asm_name(&asm_name)`, so the modifier is on the `AotInstance` itself